src/user-guide/writing-workflows/optional-outputs.rst

```diff
@@ -0,0 +1,376 @@
+      style = "dashed"
+
+      ugly
+   }
+   showdown -> good
+   showdown -> bad
+   showdown -> ugly
+   good -> fin [arrowhead="onormal"]
+   bad -> fin [arrowhead="onormal"]
+   ugly -> fin [arrowhead="onormal"]
+
+
+Cylc can't know if custom outputs are mutually exclusive or not, however. If
+they are not exclusive, the paths will be concurrent rather than alternate:
+
+For branching on custom outputs you can use an :term:`artificial dependency` to
+ensure that at least one branch executes. For the example above:
+
+.. code-block:: cylc-graph
+
+    start => showdown
+    # branch the graph depending on the outcome of "showdown"
+    showdown:good? => good
+    showdown:bad? => bad
+    showdown:ugly? => ugly
+    # join the graph back together
+    good | bad | ugly => fin
+    # ensure at least one branch is run
+    start => fin  # artificial dependency
+
+
+Family Triggers
+---------------
+
+.. (taken from https://github.com/cylc/cylc-flow/pull/4343#issuecomment-913901972)
+
+Family triggers are based on family pseudo outputs such as ``FAM:succeed-all``
+and ``FAM:fail-any`` that are short for logical expressions involving the
+corresponding member task outputs.
+
+If the member outputs are not singled out explicitly elsewhere in the graph,
+then they default to expected.
+
+For example, if ``f1`` and ``f2`` are members of ``FAM``, then this:
+
+.. code-block:: cylc-graph
+
+   FAM:fail-all => a
+
+
+means:
+
+.. code-block:: cylc-graph
+
+   f1:fail & f2:fail => a  # f1:fail and f2:fail are expected
+
+
+and this:
+
+.. code-block:: cylc-graph
+
+   FAM:succeed-any => a
+
+
+means:
+
+.. code-block:: cylc-graph
+
+   f1 & f2 => a  # f1:succeed and f2:succeed are expected
+
+
+However, the family default can be changed to optional by using ``?`` on the
+family trigger. So this:
+
+.. code-block:: cylc-graph
+
+   FAM:fail-all? => a
+
+
+means this:
+
+.. code-block:: cylc-graph
+
+   f1:fail? & f2:fail? => a  # f1:fail and f2:fail are optional
+
+
+If particular member tasks are singled out elsewhere in the graph, that
+overrides the family default for expected/optional outputs:
+
+.. code-block:: cylc-graph
+
+   # f1:fail is expected, and f2:fail is optional:
+   FAM:fail-all => a
+   f2:fail? => b
+
+
+Family Finish Triggers
+----------------------
+
+Like task ``:finish`` triggers, family ``:finish-all/any`` triggers are
+different because ``:finish`` is a pseudo output involving both ``:succeed``
+and ``:fail``, which are mutually exclusive outputs that must both be optional
+if both are used.
+
+Also like task ``:finish`` triggers, use of ``?`` is illegal on a family
+finish trigger, because the underlying member outputs must already be optional.
+
+.. code-block:: cylc-graph
+
+   FAM:finish-all => a  # f1:succeed/fail and f2:succeed/fail are optional
+   FAM:finish-any => a  # (ditto)
+
+   FAM:finish-all? => b  # ERROR
+
+
+.. note::
+   Cylc 8 does not :term:`suicide triggers <suicide trigger>` to remove tasks
```

Comment:
```suggestion
   Cylc 8 does not need :term:`suicide triggers <suicide trigger>` to remove tasks
```

Reply:
(Also fixed in the aforementioned unpushed commits)

Reply:
(fixed by the late commits)